src/etc/htmldocck.py

```diff
@@ -241,11 +241,236 @@ def concat_multi_lines(f):
 
 
 LINE_PATTERN = re.compile(r'''
-    (?<=(?<!\S))(?P<invalid>!?)@(?P<negated>!?)
-    (?P<cmd>[A-Za-z]+(?:-[A-Za-z]+)*)
+    //@\s+
+    (?P<negated>!?)(?P<cmd>[A-Za-z]+(?:-[A-Za-z]+)*)
     (?P<args>.*)$
 ''', re.X | re.UNICODE)
 
+# Equivalent to `src/tools/compiletest/src/header.rs` constant of the same name.
+KNOWN_DIRECTIVE_NAMES = [
```

Comment:
Instead of having to maintain a duplicate of this list, could you instead `include!` it or `#[path] mod` it (which might necessitate splitting stuff out of `header.rs`, I'm on mobile can't check)?

Could you add a FIXME to both compiletest and htmldocck that this setup (the double-parsing essentially) is temporary(tm) until we figure out the child framework story (registering hooks etc.)? Thanks!